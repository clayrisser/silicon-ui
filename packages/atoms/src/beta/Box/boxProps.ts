import { ReactNode } from 'react';
import { ThemedStyledProps } from 'native-theme-ui';
import {
  GestureResponderEvent,
  PanResponderGestureState,
  TouchableOpacityProps as NativeTouchableOpacityProps,
  ViewProps
} from 'react-native';
import { createSplitProps } from '../../util';

export interface BoxProps
  extends CustomBoxProps,
    StyledNativeViewProps,
    TouchableOpacityProps,
    TouchableProps {}

export interface CustomBoxProps {
  autoContrast?: boolean | 'A' | 'AA' | 'AAA';
  children?: ReactNode;
  releasePressOnExit?: boolean;
}

export interface StyledNativeViewProps
  extends Omit<ThemedStyledProps<ViewProps>, 'ref'> {}

export type TouchableOpacityProps = Omit<
  NativeTouchableOpacityProps,
  'onPress' | 'onPressIn' | 'onPressOut'
>;

export interface TouchableProps {
  onPress?: (
    e: GestureResponderEvent | React.MouseEvent<HTMLDivElement, MouseEvent>,
    gestureState?: PanResponderGestureState
  ) => void;
  onPressOut?: (
    e: GestureResponderEvent | React.MouseEvent<HTMLDivElement, MouseEvent>,
    gestureState?: PanResponderGestureState
  ) => void;
  onPressIn?: (
    e: GestureResponderEvent | React.MouseEvent<HTMLDivElement, MouseEvent>,
    gestureState?: PanResponderGestureState
  ) => void;
  onPull?: (
    e: GestureResponderEvent | React.MouseEvent<HTMLDivElement, MouseEvent>,
    gestureState?: PanResponderGestureState
  ) => void;
}

interface SplitProps {
  customBoxProps: CustomBoxProps;
  styledNativeViewProps: StyledNativeViewProps;
  touchableOpacityProps: TouchableOpacityProps;
  touchableProps: TouchableProps;
}

export const splitProps = createSplitProps<BoxProps, SplitProps>(
  {
    customBoxProps: ['autoContrast', 'children', 'releasePressOnExit'],
    touchableProps: ['onPress', 'onPressIn', 'onPressOut', 'onPull'],
    touchableOpacityProps: [
      'activeOpacity',
      'delayLongPress',
      'delayPressIn',
      'delayPressOut',
      'disabled',
      'hitSlop',
      'onBlur',
      'onFocus',
      'onLayout',
      'onLongPress',
      'pressRetentionOffset',
      'testID'
    ]
  },
  'styledNativeViewProps'
);
