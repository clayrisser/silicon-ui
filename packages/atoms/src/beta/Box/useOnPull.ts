import { useState, useRef, RefObject } from 'react';
import {
  GestureResponderEvent,
  GestureResponderHandlers,
  NativeMethods,
  PanResponder,
  PanResponderGestureState
} from 'react-native';

export default function useOnPull({
  boxRef,
  onPress,
  onPressIn,
  onPressOut,
  onPull,
  releasePressOnExit
}: UseOnPullOptions): { panHandlers: GestureResponderHandlers | undefined } {
  const [initialPosition, setInitialPosition] = useState<Position>([0, 0]);
  const [pressed, setPressed] = useState(false);

  async function exitedBox(
    e: GestureResponderEvent,
    gestureState: PanResponderGestureState
  ) {
    const { locationX, locationY } = e.nativeEvent;
    const { dx, dy } = gestureState;
    const [initialX, initialY] = initialPosition;
    const [x, y] = [
      ...(Math.abs(dx) > initialX && dx < 0 ? [-locationX] : [locationX]),
      ...(Math.abs(dy) > initialY && dy < 0 ? [-locationY] : [locationY])
    ];
    const [width, height] = await new Promise<Position>((resolve) => {
      boxRef?.current?.measure(
        (_width: number, _height: number, fx: number, fy: number) => {
          resolve([fx, fy]);
        }
      );
    });
    return x < 0 || y < 0 || x > width || y > height;
  }

  const panResponder = onPull
    ? useRef(
        PanResponder.create({
          onMoveShouldSetPanResponder: () => pressed,
          onStartShouldSetPanResponder: () => true,
          onPanResponderMove: async (
            e: GestureResponderEvent,
            gestureState: PanResponderGestureState
          ) => {
            if (!pressed) return;
            e.persist();
            if (releasePressOnExit && (await exitedBox(e, gestureState))) {
              // pressed = false;
              setPressed(false);
              if (onPressOut) onPressOut(e, gestureState);
              if (onPress) onPress(e, gestureState);
            } else if (onPull) {
              onPull(e, gestureState);
            }
          },
          onPanResponderGrant: (
            e: GestureResponderEvent,
            gestureState: PanResponderGestureState
          ) => {
            const { locationX, locationY } = e.nativeEvent;
            // initialPosition = [locationX, locationY];
            setInitialPosition([locationX, locationY]);
            // pressed = true;
            setPressed(true);
            if (onPressIn) onPressIn(e, gestureState);
          },
          onPanResponderRelease: (
            e: GestureResponderEvent,
            gestureState: PanResponderGestureState
          ) => {
            // initialPosition = [0, 0];
            setInitialPosition([0, 0]);
            if (pressed) {
              if (onPressOut) onPressOut(e, gestureState);
              if (onPress) onPress(e, gestureState);
            }
            // pressed = false;
            setPressed(false);
          }
        })
      )
    : { current: { panHandlers: {} } };
  const { panHandlers } = panResponder.current;
  return { panHandlers };
}

export interface UseOnPullOptions {
  boxRef: RefObject<NativeMethods>;
  releasePressOnExit?: boolean;
  onPull?: (
    e: GestureResponderEvent,
    gestureState?: PanResponderGestureState
  ) => void;
  onPressIn?: (
    e: GestureResponderEvent,
    gestureState?: PanResponderGestureState
  ) => void;
  onPress?: (
    e: GestureResponderEvent,
    gestureState?: PanResponderGestureState
  ) => void;
  onPressOut?: (
    e: GestureResponderEvent,
    gestureState?: PanResponderGestureState
  ) => void;
}

export type Position = [number, number];
