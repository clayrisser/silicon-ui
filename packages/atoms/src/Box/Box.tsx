import styled, { StyledComponent } from '@emotion/styled';
import React, {
  DetailedHTMLProps,
  HTMLAttributes,
  LegacyRef,
  forwardRef,
  useEffect,
  useState
} from 'react';
import {
  background,
  border,
  color,
  compose,
  flexbox,
  layout,
  position,
  shadow,
  space,
  typography
} from 'styled-system';
import { BoxProps, splitProps } from './boxProps';

export type DetailedHTMLDivProps = DetailedHTMLProps<
  HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

const HTMLDiv: StyledComponent<
  DetailedHTMLDivProps,
  BoxProps,
  object
> = styled.div(
  compose(
    background,
    border,
    color,
    flexbox,
    layout,
    position,
    shadow,
    space,
    typography
  )
);

const Box = forwardRef((props: BoxProps, boxRef: LegacyRef<any>) => {
  let [entered, setEntered] = useState(true);
  let [pressed, setPressed] = useState(false);
  const { styledBoxProps, customBoxProps, touchableOpacityProps } = splitProps(
    props
  );

  useEffect(() => {
    if (!props.onPressOut && !props.onPull) return () => {};
    function handleMouseUp(e: Event) {
      if (!entered && pressed && props.onPressOut) {
        props.onPressOut(bakeEvent(e));
      }
      pressed = false;
      setPressed(false);
    }
    function handleMouseMove(e: Event) {
      if (!entered && pressed && props.onPull) {
        props.onPull(bakeEvent(e));
      }
    }
    window.document.body.addEventListener('mouseup', handleMouseUp);
    window.document.body.addEventListener('mousemove', handleMouseMove);
    return () => {
      window.document.body.removeEventListener('mouseup', handleMouseUp);
      window.document.body.removeEventListener('mousemove', handleMouseMove);
    };
  }, [entered, pressed]);

  function handleMouseEnter() {
    entered = true;
    setEntered(entered);
  }

  function handleMouseLeave(e: React.MouseEvent<HTMLDivElement, MouseEvent>) {
    if (props.releasePressOnExit) {
      if (pressed && props.onPressOut) props.onPressOut(e);
      pressed = false;
      setPressed(false);
    }
    entered = false;
    setEntered(entered);
  }

  function handleClick(e: any) {
    if (props.onPress) props.onPress(e);
  }

  function handleMouseMove(e: React.MouseEvent<HTMLDivElement, MouseEvent>) {
    if (pressed && props.onPull) props.onPull(e);
  }

  function handleMouseUp(e: React.MouseEvent<HTMLDivElement, MouseEvent>) {
    if (props.onPressOut) props.onPressOut(e);
    pressed = false;
    setPressed(false);
  }

  function handleMouseDown(e: React.MouseEvent<HTMLDivElement, MouseEvent>) {
    pressed = true;
    setPressed(true);
    if (props.onPressIn) props.onPressIn(e);
  }

  function handleBlur(e: React.MouseEvent<HTMLDivElement, MouseEvent>) {
    if (props.onPress) props.onPress(e);
  }

  return (
    <HTMLDiv
      {...touchableOpacityProps}
      {...(styledBoxProps as any)}
      onBlur={handleBlur}
      onClick={handleClick}
      onMouseDown={handleMouseDown}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      onMouseMove={handleMouseMove}
      onMouseUp={handleMouseUp}
      ref={boxRef}
      theme={customBoxProps.theme}
    >
      {customBoxProps.children}
    </HTMLDiv>
  );
});

function bakeEvent(e: any): React.MouseEvent<HTMLDivElement, MouseEvent> {
  return {
    pageX: e.pageX,
    pageY: e.pageY,
    target: e.target
  } as React.MouseEvent<HTMLDivElement, MouseEvent>;
}

Box.defaultProps = {
  activeOpacity: 1,
  backgroundColor: 'transparent',
  children: <></>,
  color: 'red',
  fontFamily: 'body',
  fontSize: 'body',
  fontWeight: 'body',
  height: '100%',
  lineHeight: 'body'
};

export default styled(Box)`
  :active {
    opacity: ${({ activeOpacity }: BoxProps) => activeOpacity};
  }
`;
