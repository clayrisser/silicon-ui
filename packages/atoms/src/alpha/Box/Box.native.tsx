import React, { forwardRef, Ref, useRef, useEffect } from 'react';
import useMergedRef from '@react-hook/merged-ref';
import { styled, View as DripsyView } from 'native-theme-ui';
import {
  Animated,
  NativeMethods,
  Text as NativeText,
  TouchableOpacity,
  ViewProps
} from 'react-native';
import {
  background,
  border,
  color,
  compose,
  flexbox,
  layout,
  position,
  shadow,
  space,
  typography
} from 'styled-system';
import useOnPull from './useOnPull';
import useThemeLookup from '../../hooks/useThemeLookup';
import { BoxProps, StyledBoxProps, splitProps } from './boxProps';

export const StyledView = styled<StyledBoxProps, ViewProps>(DripsyView, {
  isThemed: true
})(
  compose(
    background,
    border,
    color,
    flexbox,
    layout,
    position,
    shadow,
    space,
    typography
  )
);

export type Position = [number, number];

const Box = forwardRef<NativeMethods, BoxProps>(
  (props: BoxProps, forwardedRef: Ref<NativeMethods>) => {
    const boxRef = useRef<NativeMethods>(null);
    const themeLookup = useThemeLookup();
    const { panHandlers } = useOnPull({
      boxRef,
      onPress: props.onPress,
      onPressIn: props.onPressIn,
      onPressOut: props.onPressOut,
      onPull: props.onPull,
      releasePressOnExit: props.releasePressOnExit
    });
    const mergedRef = useMergedRef<any>(forwardedRef, boxRef);
    const {
      customBoxProps,
      nativeBoxProps,
      nativeTouchableOpacityProps,
      styledBoxProps,
      touchableOpacityProps
    } = splitProps({
      ...props,
      backgroundColor: themeLookup<string>(
        'backgroundColor',
        props.backgroundColor
      )
    });

    useEffect(() => {
      if (props.onPull && !props.backgroundColor) {
        console.warn('backgroundColor prop required for onPull to work');
      }
    }, []);

    const children =
      typeof customBoxProps.children === 'string' ? (
        <NativeText>{customBoxProps.children}</NativeText>
      ) : (
        customBoxProps.children
      );

    function isTouchable() {
      return (
        Object.keys(touchableOpacityProps).length ||
        Object.keys(nativeTouchableOpacityProps).length
      );
    }

    function renderTouchable() {
      if (nativeBoxProps.onPull) {
        return (
          <>
            <Animated.View
              {...panHandlers}
              style={{
                backgroundColor: styledBoxProps.backgroundColor as any,
                height: styledBoxProps.height as any,
                position: 'absolute',
                width: styledBoxProps.width as any
              }}
            />
            {children}
          </>
        );
      }
      if (!touchableOpacityProps.activeOpacity) {
        touchableOpacityProps.activeOpacity = 1;
      }
      return (
        <TouchableOpacity
          {...touchableOpacityProps}
          {...nativeTouchableOpacityProps}
          style={{
            height: '100%',
            width: '100%'
          }}
        >
          {children}
        </TouchableOpacity>
      );
    }

    return (
      <StyledView ref={mergedRef} {...styledBoxProps} {...nativeBoxProps}>
        {isTouchable() ? renderTouchable() : children}
      </StyledView>
    );
  }
);

Box.defaultProps = {
  color: 'text',
  fontFamily: 'body',
  fontSize: 1,
  fontWeight: 'body'
};

export default Box;
