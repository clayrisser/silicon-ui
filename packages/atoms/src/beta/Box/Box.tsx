import React, { forwardRef, Ref, useRef, useEffect } from 'react';
import { SxStyleProp } from 'theme-ui';
import useMergedRef from '@react-hook/merged-ref';
import {
  styled,
  View as ThemedView,
  Text as ThemedText
} from 'native-theme-ui';
import {
  Animated,
  NativeMethods,
  TouchableOpacity,
  ViewProps
} from 'react-native';
import useOnPull from './useOnPull';
import useThemeLookup from '../../hooks/useThemeLookup';
import { BoxProps, splitProps } from './boxProps';
import useColor from '../../hooks/useColor';

const StyledNativeView = styled<ViewProps>(ThemedView, { isThemed: true })({});

const defaultSx: SxStyleProp = {
  color: 'text',
  fontFamily: 'body',
  fontSize: 1,
  fontWeight: 'body',
  position: 'inherit'
};

export type Position = [number, number];

const Box = forwardRef<NativeMethods, BoxProps>(
  (props: BoxProps, forwardedRef: Ref<NativeMethods>) => {
    const themeLookup = useThemeLookup();
    const {
      customBoxProps,
      styledNativeViewProps,
      sx,
      touchableOpacityProps,
      touchableProps
    } = splitProps(props, defaultSx);
    const color = useColor(props, sx);
    const backgroundColor = props.onPull
      ? themeLookup<string>('backgroundColor', sx.backgroundColor || '#FFFFFF')
      : sx.backgroundColor;
    const boxRef = useRef<NativeMethods>(null);
    const { panHandlers } = useOnPull({
      boxRef,
      onPress: touchableProps.onPress,
      onPressIn: touchableProps.onPressIn,
      onPressOut: touchableProps.onPressOut,
      onPull: touchableProps.onPull,
      releasePressOnExit: customBoxProps.releasePressOnExit
    });
    const mergedRef = useMergedRef<any>(forwardedRef, boxRef);

    useEffect(() => {
      if (touchableProps.onPull && !sx.backgroundColor) {
        console.warn('backgroundColor required for onPull to work');
      }
    }, []);

    function renderChildren() {
      const { children } = customBoxProps;
      if (typeof children === 'string') {
        return <ThemedText sx={{ color }}>{children}</ThemedText>;
      }
      return children;
    }

    function isTouchable() {
      return (
        Object.keys(touchableOpacityProps).length ||
        Object.keys(touchableProps).length
      );
    }

    function renderTouchable() {
      if (touchableProps.onPull) {
        return (
          <>
            <Animated.View
              {...panHandlers}
              style={{
                backgroundColor: backgroundColor as any,
                height: sx.height as any,
                position: 'absolute',
                width: sx.width as any
              }}
            />
            {renderChildren()}
          </>
        );
      }
      if (!touchableOpacityProps.activeOpacity) {
        touchableOpacityProps.activeOpacity = 1;
      }
      return (
        <TouchableOpacity
          {...touchableOpacityProps}
          {...touchableProps}
          style={{
            height: '100%',
            width: '100%'
          }}
        >
          {renderChildren()}
        </TouchableOpacity>
      );
    }

    return (
      <StyledNativeView ref={mergedRef} {...styledNativeViewProps} sx={sx}>
        {isTouchable() ? renderTouchable() : renderChildren()}
      </StyledNativeView>
    );
  }
);

Box.defaultProps = {};

export default Box;
