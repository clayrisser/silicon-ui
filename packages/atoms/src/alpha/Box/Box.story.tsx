import React, { FC } from 'react';
import { action } from '@storybook/addon-actions';
import { invert } from '@theme-ui/color';
import { useThemeUI } from 'theme-ui';
import { withKnobs, text, number, color } from '@storybook/addon-knobs';
import Box from './Box';
import Wrapper from '../../../storybook/Wrapper';
import docs from './docs';
import storiesOf from '../../../storybook/storiesOf';
import useThemeLookup from '../../hooks/useThemeLookup';
import withThemeProvider from '../../../storybook/withThemeProvider';

export const BoxStory: FC = () => {
  const themeLookup = useThemeLookup();
  const { theme } = useThemeUI();
  const backgroundColor = 'primary';
  let textColor = theme.colors?.text || ('text' as string);
  try {
    textColor = invert(themeLookup<string>('backgroundColor', backgroundColor))(
      theme
    );
  } catch (err) {}
  return (
    <Wrapper>
      <Box
        // onPull={action('onPull')}
        backgroundColor={color('backgroundColor', backgroundColor)}
        borderBottomColor={color('border-bottom-color', '')}
        borderBottomLeftRadius={number('border-left-radius', 0)}
        borderBottomRightRadius={number('border-right-radius', 0)}
        borderBottomWidth={number('border-bottom-width', 0)}
        borderColor={color('border-color', '')}
        borderLeftColor={color('border-left-color', '')}
        borderLeftWidth={number('border-left-width', 0)}
        borderRadius={number('borderRadius', 0)}
        borderRightColor={color('border-right-color', '')}
        borderRightWidth={number('border-right-width', 0)}
        borderStyle={text('border-style', 'solid')}
        borderTopColor={color('border-top-color', '')}
        borderTopLeftRadius={number('borderTop-leftRadius', 0)}
        borderTopRightRadius={number('borderTop-RightRadius', 0)}
        borderTopWidth={number('border-top-width', 0)}
        borderWidth={number('border-width', 0)}
        color={color('color', textColor)}
        height={number('height', 200)}
        marginBottom={number('margin-bottom', 0)}
        marginLeft={number('margin-left', 0)}
        marginRight={number('margin-right', 0)}
        marginTop={number('margin-top', 0)}
        maxHeight={number('max-height', 999)}
        minHeight={number('min-height', 0)}
        onPress={action('onPress')}
        onPressIn={action('onPressIn')}
        onPressOut={action('onPressOut')}
        opacity={number('opacity', 1)}
        padding={number('padding', 0)}
        paddingBottom={number('padding-bottom', 0)}
        paddingLeft={number('padding-left', 0)}
        paddingRight={number('padding-right', 0)}
        paddingTop={number('padding-top', 0)}
        width={number('width', 200)}
        zIndex={number('z-index', 0)}
      >
        {text('children', 'data')}
      </Box>
    </Wrapper>
  );
};

storiesOf('Alpha/Box', module)
  .addDecorator(withKnobs)
  .addDecorator(withThemeProvider)
  .addParameters({
    docs: { page: docs }
    // jest: ['Button.test.tsx']
  })
  .add('with knobs', () => <BoxStory />);
