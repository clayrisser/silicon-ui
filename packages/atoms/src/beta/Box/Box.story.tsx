import React, { FC } from 'react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import Box from './Box';
import Wrapper from '../../../storybook/Wrapper';
import docs from './docs';
import storiesOf from '../../../storybook/storiesOf';
import withThemeProvider from '../../../storybook/withThemeProvider';

export const BoxStory: FC = () => {
  return (
    <Wrapper>
      <Box
        onPull={action('onPull')}
        sx={{ backgroundColor: text('backgroundColor', 'background') }}
        onFocus={action('onFocus')}
        onPress={action('onPress')}
        onPressIn={action('onPressIn')}
        onPressOut={action('onPressOut')}
        autoContrast={select<'A' | 'AA' | 'AAA'>(
          'autoContrast',
          {
            false: '' as 'A',
            A: 'A',
            AA: 'AA',
            AAA: 'AAA'
          },
          'AA'
        )}
      >
        {text('children', 'data')}
      </Box>
    </Wrapper>
  );
};

storiesOf('Beta/Box', module)
  .addDecorator(withKnobs)
  .addDecorator(withThemeProvider)
  .addParameters({
    docs: { page: docs }
    // jest: ['Button.test.tsx']
  })
  .add('with knobs', () => <BoxStory />);
