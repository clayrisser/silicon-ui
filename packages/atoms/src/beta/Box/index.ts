import Box from './Box';
import { BoxProps } from './boxProps';

export { BoxProps };
export default Box;
