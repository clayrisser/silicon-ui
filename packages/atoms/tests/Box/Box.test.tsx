import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Box from '../../src/Box/Box';

describe('<Box />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Box />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Box />', () => {
  it('renders correctly', () => {
    const fontSize = 16;
    const bg = 'white';
    const color = 'black';
    const height = 90;
    const padding = 9;
    const width = 600;
    const border = 1;
    const borderStyle = 'solid';
    const borderColor = 'red';
    const backgroundColor = 'transparent';
    const activeOpacity = 1;
    const tree = renderer
      .create(
        <Box
          fontSize={fontSize}
          bg={bg}
          color={color}
          height={height}
          p={padding}
          border={border}
          width={width}
          borderStyle={borderStyle}
          borderColor={borderColor}
          backgroundColor={backgroundColor}
          activeOpacity={activeOpacity}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('handles onClick', () => {
  //   const handleClick = jest.fn();
  //   const handleMouseDown = jest.fn();
  //   const handleMouseLeave = jest.fn();
  //   const handleMouseMove = jest.fn();
  //   const handleMouseUp = jest.fn();
  //   const handleBlur = jest.fn();
  //   const { container } = render(
  //     <Box
  //       onClick={handleClick}
  //       onBlur={handleBlur}
  //       onMouseEnter={handleMouseDown}
  //       onMouseDown={handleMouseDown}
  //       onMouseLeave={handleMouseLeave}
  //       onMouseMove={handleMouseMove}
  //       onMouseUp={handleMouseUp}
  //     />
  //   );
  //   fireEvent.click(container.querySelector('HTMLDiv')!);
  //   expect(handleClick).toHaveBeenCalledTimes(1);
  //   fireEvent.mouseDown(container.querySelector('HTMLDiv')!);
  //   expect(handleMouseDown).toBe(undefined);
  // });
});
