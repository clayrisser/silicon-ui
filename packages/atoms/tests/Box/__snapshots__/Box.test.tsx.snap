// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders correctly 1`] = `
<div
  className="css-12p9kah"
  color="red"
  fontFamily="body"
  fontSize="body"
  fontWeight="body"
  height="100%"
  onBlur={[Function]}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  width="100%"
/>
`;
